ds3-sdk-integration/src/test/java/com/spectralogic/ds3client/integration/PutJobManagement_Test.java

```diff
@@ -0,0 +1,600 @@
+
+            assertThat(responseAfterModify.getMasterObjectListResult().getStartDate(), is(newDate));
+
+        } finally {
+            deleteAllContents(client, BUCKET_NAME);
+        }
+    }
+
+    @Test
+    public void cancelJob() throws IOException, SignatureException, XmlProcessingException {
+        
+        try {
+            final Ds3ClientHelpers.Job job =
+                    HELPERS.startWriteJob(BUCKET_NAME, Lists.newArrayList(new Ds3Object("testOne", 2)));
+
+            final CancelJobSpectraS3Response response = client
+                    .cancelJobSpectraS3(new CancelJobSpectraS3Request(job.getJobId()));
+            assertEquals(response.getStatusCode(),204);
+
+            assertTrue(client.getActiveJobsSpectraS3(new GetActiveJobsSpectraS3Request())
+                    .getActiveJobListResult().getActiveJobs().isEmpty());
+
+        } finally {
+            deleteAllContents(client, BUCKET_NAME);
+        }
+    }
+
+    @Test
+    public void clearAllCanceledJobs() throws IOException, SignatureException, XmlProcessingException {
+
+        try {
+            final Ds3ClientHelpers.Job job =
+                    HELPERS.startWriteJob(BUCKET_NAME, Lists.newArrayList(new Ds3Object("testOne", 2)));
+            final CancelJobSpectraS3Response response = client
+                    .cancelJobSpectraS3(new CancelJobSpectraS3Request(job.getJobId()));
+            client.clearAllCanceledJobsSpectraS3(new ClearAllCanceledJobsSpectraS3Request());
+            final List canceledJobsList = client.
+                    getCanceledJobsSpectraS3(new GetCanceledJobsSpectraS3Request())
+                    .getCanceledJobListResult().getCanceledJobs();
+
+            assertTrue(canceledJobsList.isEmpty());
+
+        } finally {
+            deleteAllContents(client, BUCKET_NAME);
+        }
+    }
+
+    @Test
+    public void truncateJobCancelWithOutForce() throws IOException, SignatureException, XmlProcessingException, URISyntaxException, InterruptedException {
+
+        final int testTimeOutSeconds = 5;
+        final String book1 = "beowulf.txt";
+        final Path objPath1 = ResourceUtils.loadFileResource(RESOURCE_BASE_NAME + book1);
+        final Ds3Object obj1 = new Ds3Object(book1, Files.size(objPath1));
+        final Ds3Object obj2 = new Ds3Object("place_holder", 5000000);
+
+        try {
+            final Ds3ClientHelpers.Job putJob = HELPERS.startWriteJob(BUCKET_NAME, Lists.newArrayList(obj1, obj2));
+            final UUID jobId = putJob.getJobId();
+            final SeekableByteChannel book1Channel = new ResourceObjectPutter(RESOURCE_BASE_NAME).buildChannel(book1);
+            client.putObject(new PutObjectRequest(BUCKET_NAME, book1, book1Channel, jobId, 0, Files.size(objPath1)));
+
+            //make sure black pearl has updated it's job to show 1 object in cache
+            final long startTime = System.nanoTime();
+            long cachedSize = 0;
+            while (cachedSize == 0) {
```

Comment:
It looks like my comment about this section of code got lost.  Please pull this check out into it's own method since it's being reused in several places.